Importo las librerias necesarias para el manejo de los dataset

In [2]:
import pandas as pd
import numpy as np

Importo lobreria de google y monto disco de Google Drive, seleccionando carpeta donde se encuentra el Notebook y CSV

In [3]:
from google.colab import drive
import os
drive.mount('/content/gdrive')
%cd '/content/gdrive/MyDrive/Coder_DS_CristianSegovia/'

Mounted at /content/gdrive
/content/gdrive/MyDrive/Coder_DS_CristianSegovia


## ------ PRIMER DATASET ------

### -- Levanto .csv desde Google Drive y genero dataset "df_ventas", valido carga inicial con un .head()


In [4]:
df_ventas = pd.read_csv('VentasDiciembre.csv',sep=',')
df_ventas.head()

PARTITION_DATE  LOC_ID    PROD_ID CLASE_ID SUBCLASE_ID LINEA_ID  \
0  2022-12-01 00:00:00 UTC    2000  882570648  J010503   J01050303      J01   
1  2022-12-01 00:00:00 UTC    2000  882644750  J040705   J04070501      J04   
2  2022-12-01 00:00:00 UTC    2000  882522925  J120105   J12010504      J12   
3  2022-12-01 00:00:00 UTC    2000  881530356  J030101   J03010102      J03   
4  2022-12-01 00:00:00 UTC    2000  881704162  J050302   J05030201      J05   

   POOL_ID               POOL     ORIGEN ATT_TIPO_MARCA  \
0      1.0  BLANDOS VESTUARIO  IMPORTADO         Propia   
1      1.0  BLANDOS VESTUARIO  IMPORTADO         Propia   
2      3.0       BLANDOS DECO  IMPORTADO         Propia   
3      1.0  BLANDOS VESTUARIO  IMPORTADO         Propia   
4      1.0  BLANDOS VESTUARIO  IMPORTADO         Propia   

                            MUNDO_TIENDA MUNDO_LOGISTICA  TRAN_TOTAL_AMT  \
0                           SPORT HOMBRE  REGULAR BLANDO            7980   
1                            MUJER JOVEN         COLGADO           11990   
2                           ROPA DE CAMA  REGULAR BLANDO           39960   
3  VESTUARIO DEPORTIVO HOMBRE Y FOOTBALL  REGULAR BLANDO           20990   
4                COLECCION JUVENIL MUJER  REGULAR BLANDO            5990   

   TRAN_QTY  PRECIO_NORMAL  COSTO_PROMEDIO  COSTO_REPOSICION  \
0         2           6990         2312.36           2454.29   
1         1          16990         5037.48           4772.20   
2         4          19990         4989.18           4947.60   
3         1          29990         9859.13          12096.97   
4         1           9990         3149.97           2993.98   

   COSTO_LOGISTICO_MEDIO  
0            2604.511063  
1            2604.511063  
2            2965.198401  
3            2965.198401  
4            2604.511063

###-- Informacion del archivo CSV utilizado:

Este archivo es una extraccion de informacion de ventas de una empresa de retail para el pediodo Diciembre 2022 (diario), la informacion viene detallada a nivel de SKU (PROD_ID) y cuenta con caracteristicas del producto y detalles de la Venta, costos de reposicion y costo logistico estimado. 

###-- Revisiones de dataset "df_ventas"

In [ ]:
print(df_ventas.shape)
df_ventas.head()

(2500051, 18)


PARTITION_DATE  LOC_ID    PROD_ID CLASE_ID SUBCLASE_ID LINEA_ID  \
0  2022-12-01 00:00:00 UTC    2000  882570648  J010503   J01050303      J01   
1  2022-12-01 00:00:00 UTC    2000  882644750  J040705   J04070501      J04   
2  2022-12-01 00:00:00 UTC    2000  882522925  J120105   J12010504      J12   
3  2022-12-01 00:00:00 UTC    2000  881530356  J030101   J03010102      J03   
4  2022-12-01 00:00:00 UTC    2000  881704162  J050302   J05030201      J05   

   POOL_ID               POOL     ORIGEN ATT_TIPO_MARCA  \
0      1.0  BLANDOS VESTUARIO  IMPORTADO         Propia   
1      1.0  BLANDOS VESTUARIO  IMPORTADO         Propia   
2      3.0       BLANDOS DECO  IMPORTADO         Propia   
3      1.0  BLANDOS VESTUARIO  IMPORTADO         Propia   
4      1.0  BLANDOS VESTUARIO  IMPORTADO         Propia   

                            MUNDO_TIENDA MUNDO_LOGISTICA  TRAN_TOTAL_AMT  \
0                           SPORT HOMBRE  REGULAR BLANDO            7980   
1                            MUJER JOVEN         COLGADO           11990   
2                           ROPA DE CAMA  REGULAR BLANDO           39960   
3  VESTUARIO DEPORTIVO HOMBRE Y FOOTBALL  REGULAR BLANDO           20990   
4                COLECCION JUVENIL MUJER  REGULAR BLANDO            5990   

   TRAN_QTY  PRECIO_NORMAL  COSTO_PROMEDIO  COSTO_REPOSICION  \
0         2           6990         2312.36           2454.29   
1         1          16990         5037.48           4772.20   
2         4          19990         4989.18           4947.60   
3         1          29990         9859.13          12096.97   
4         1           9990         3149.97           2993.98   

   COSTO_LOGISTICO_MEDIO  
0            2604.511063  
1            2604.511063  
2            2965.198401  
3            2965.198401  
4            2604.511063

In [6]:
#Total de registros
print(df_ventas[['TRAN_TOTAL_AMT']].count())

# contar el numero de nulos de ventas por columnas
ventas_missing = (df_ventas[['TRAN_TOTAL_AMT']] == 0).sum()
print(ventas_missing)

TRAN_TOTAL_AMT    2500051
dtype: int64
TRAN_TOTAL_AMT    633628
dtype: int64


In [16]:
# Estadistica descriptiva de las columasnde Venta y Unidades Vendidas
df_ventas[['TRAN_TOTAL_AMT','TRAN_QTY']].describe()

TRAN_TOTAL_AMT      TRAN_QTY
count    2.500051e+06  2.500051e+06
mean     2.823554e+04  9.230120e-01
std      1.006521e+05  6.237302e-01
min     -5.299990e+06 -2.000000e+01
25%      0.000000e+00  1.000000e+00
50%      1.099000e+04  1.000000e+00
75%      2.499000e+04  1.000000e+00
max      1.021986e+07  6.000000e+01

## ------ SEGUNDO DATASET ------

### -- Levanto .csv desde GITHUB y genero dataset "df_sismos", valido carga inicial con un .head()

In [ ]:
df_sismos = pd.read_csv('https://raw.githubusercontent.com/CristianSegoviaMontier/Dataset_DS_Coder1/main/eqcat_CEUS1568-2018.csv',sep=',')
df_sismos.head()


Mw  long_degE  lat_degN  depth_km  year  month  day  hour  minute  \
0  4.85      -72.5      41.5         0  1568      1    1     0       0   
1  3.31      -72.5      41.5         0  1574      1    1     0       0   
2  3.31      -72.5      41.5         0  1584      1    1     0       0   
3  3.31      -72.5      41.5         0  1592      1    1     0       0   
4  2.68      -70.8      42.8         0  1643      3   15    12       0   

   second  mag_uncert   Mwe     N_              comment  
0     0.0       1.000  4.85  1.000  J94|wu,4.85(X_1.00)  
1     0.0       0.512  3.31  2.004  SHM|SSC|we,TMP00003  
2     0.0       0.512  3.31  2.004  SHM|SSC|we,TMP00004  
3     0.0       0.512  3.31  2.004  SHM|SSC|we,TMP00005  
4     0.0       0.500  2.68  1.940       NCE|i0,04.0WES

###-- Consulto readme.txt del creador del archivo:

Data file name: eqcat_CEUS1568-2018.txt(.csv)

Purpose: A seismicity catalog generated for a U.S. Geological Survey (USGS) and Nuclear Regulatory Commission (NRC) study.

Summary: The earthquake catalog was generated in August 2018 by Chuck Mueller (cmueller@usgs.gov; Emeritus) using the standard NSHM methodology for the CEUS (Mueller, 2019). 
Pre-existing catalogs were merged, duplicate records were removed, the catalog was declustered, and induced earthquakes were removed. 
The final catalog contains 6802 records, M2.5 to 7.8, and extends from 1568 through July 2018. The catalog contains 14 columns defined below:

Mw	nonsepecific moment magnitude
long_degE [longitude degrees east]	longitude decimal degrees east WGS84
lat_degN [latitude degrees north]	latitude decimal degrees north WGS84
depth_km	depth in kilometers
year		year
month		month
day		day
hour		hour
minute		minute
second		second
mag_uncert[magnitude uncertainty]	magnitude uncertainty
Mwe		expected moment magnitude, often, but not always, equal to Mw
N_[N*]		counting factor for rate calculations
comment	comment

References:

Mueller CS (2019) Earthquake catalogs for the USGS National Seismic Hazard Maps. Seismological Research Letters 90(1): 251-261, https://doi.org/10.1785/0220170108.


###-- Revisiones de dataset "df_sismos"

In [ ]:
print(df_sismos.shape)
df_sismos.head()

(6802, 14)


Mw  long_degE  lat_degN  depth_km  year  month  day  hour  minute  \
0  4.85      -72.5      41.5         0  1568      1    1     0       0   
1  3.31      -72.5      41.5         0  1574      1    1     0       0   
2  3.31      -72.5      41.5         0  1584      1    1     0       0   
3  3.31      -72.5      41.5         0  1592      1    1     0       0   
4  2.68      -70.8      42.8         0  1643      3   15    12       0   

   second  mag_uncert   Mwe     N_              comment  
0     0.0       1.000  4.85  1.000  J94|wu,4.85(X_1.00)  
1     0.0       0.512  3.31  2.004  SHM|SSC|we,TMP00003  
2     0.0       0.512  3.31  2.004  SHM|SSC|we,TMP00004  
3     0.0       0.512  3.31  2.004  SHM|SSC|we,TMP00005  
4     0.0       0.500  2.68  1.940       NCE|i0,04.0WES

## ------ TERCER DATASET ------

### -- Levanto .csv desde Google Drive y genero dataset "df_crashes", valido carga inicial con un .head()

In [18]:
df_crashes = pd.read_csv('Vehicle_Crashes.csv',sep=',')
df_crashes.head()

<ipython-input-18-b3297a854b24>:1: DtypeWarning: Columns (3) have mixed types. Specify dtype option on import or set low_memory=False.
  df_crashes = pd.read_csv('Vehicle_Crashes.csv',sep=',')


CRASH DATE CRASH TIME   BOROUGH ZIP CODE   LATITUDE  LONGITUDE  \
0  09/11/2021       2:39       NaN      NaN        NaN        NaN   
1  03/26/2022      11:45       NaN      NaN        NaN        NaN   
2  06/29/2022       6:55       NaN      NaN        NaN        NaN   
3  09/11/2021       9:35  BROOKLYN  11208.0  40.667202 -73.866500   
4  12/14/2021       8:13  BROOKLYN  11233.0  40.683304 -73.917274   

                  LOCATION           ON STREET NAME CROSS STREET NAME  \
0                      NaN    WHITESTONE EXPRESSWAY         20 AVENUE   
1                      NaN  QUEENSBORO BRIDGE UPPER               NaN   
2                      NaN       THROGS NECK BRIDGE               NaN   
3    (40.667202, -73.8665)                      NaN               NaN   
4  (40.683304, -73.917274)          SARATOGA AVENUE    DECATUR STREET   

           OFF STREET NAME  ...  CONTRIBUTING FACTOR VEHICLE 2  \
0                      NaN  ...                    Unspecified   
1                      NaN  ...                            NaN   
2                      NaN  ...                    Unspecified   
3  1211      LORING AVENUE  ...                            NaN   
4                      NaN  ...                            NaN   

   CONTRIBUTING FACTOR VEHICLE 3  CONTRIBUTING FACTOR VEHICLE 4  \
0                            NaN                            NaN   
1                            NaN                            NaN   
2                            NaN                            NaN   
3                            NaN                            NaN   
4                            NaN                            NaN   

   CONTRIBUTING FACTOR VEHICLE 5  COLLISION_ID  VEHICLE TYPE CODE 1  \
0                            NaN       4455765                Sedan   
1                            NaN       4513547                Sedan   
2                            NaN       4541903                Sedan   
3                            NaN       4456314                Sedan   
4                            NaN       4486609                  NaN   

   VEHICLE TYPE CODE 2  VEHICLE TYPE CODE 3 VEHICLE TYPE CODE 4  \
0                Sedan                  NaN                 NaN   
1                  NaN                  NaN                 NaN   
2        Pick-up Truck                  NaN                 NaN   
3                  NaN                  NaN                 NaN   
4                  NaN                  NaN                 NaN   

  VEHICLE TYPE CODE 5  
0                 NaN  
1                 NaN  
2                 NaN  
3                 NaN  
4                 NaN  

[5 rows x 29 columns]

###-- Consulto informacion del creador del archivo (Kaggle):

The Motor Vehicle Collisions crash table contains details on the crash event. Each row represents a crash event. The Motor Vehicle Collisions data tables contain information from all police reported motor vehicle collisions in NYC. The police report (MV104-AN) is required to be filled out for collisions where someone is injured or killed, or where there is at least $1000 worth of damagecheck here. It should be noted that the data is preliminary and subject to change when the MV-104AN forms are amended based on revised crash details.For the most accurate, up to date statistics on traffic fatalities, please refer to the NYPD Motor Vehicle Collisions page (updated weekly) or Vision Zero View (updated monthly).

###-- Revisiones de dataset "df_craches"

In [19]:
print(df_crashes.shape)
df_crashes.head()

(1972121, 29)


CRASH DATE CRASH TIME   BOROUGH ZIP CODE   LATITUDE  LONGITUDE  \
0  09/11/2021       2:39       NaN      NaN        NaN        NaN   
1  03/26/2022      11:45       NaN      NaN        NaN        NaN   
2  06/29/2022       6:55       NaN      NaN        NaN        NaN   
3  09/11/2021       9:35  BROOKLYN  11208.0  40.667202 -73.866500   
4  12/14/2021       8:13  BROOKLYN  11233.0  40.683304 -73.917274   

                  LOCATION           ON STREET NAME CROSS STREET NAME  \
0                      NaN    WHITESTONE EXPRESSWAY         20 AVENUE   
1                      NaN  QUEENSBORO BRIDGE UPPER               NaN   
2                      NaN       THROGS NECK BRIDGE               NaN   
3    (40.667202, -73.8665)                      NaN               NaN   
4  (40.683304, -73.917274)          SARATOGA AVENUE    DECATUR STREET   

           OFF STREET NAME  ...  CONTRIBUTING FACTOR VEHICLE 2  \
0                      NaN  ...                    Unspecified   
1                      NaN  ...                            NaN   
2                      NaN  ...                    Unspecified   
3  1211      LORING AVENUE  ...                            NaN   
4                      NaN  ...                            NaN   

   CONTRIBUTING FACTOR VEHICLE 3  CONTRIBUTING FACTOR VEHICLE 4  \
0                            NaN                            NaN   
1                            NaN                            NaN   
2                            NaN                            NaN   
3                            NaN                            NaN   
4                            NaN                            NaN   

   CONTRIBUTING FACTOR VEHICLE 5  COLLISION_ID  VEHICLE TYPE CODE 1  \
0                            NaN       4455765                Sedan   
1                            NaN       4513547                Sedan   
2                            NaN       4541903                Sedan   
3                            NaN       4456314                Sedan   
4                            NaN       4486609                  NaN   

   VEHICLE TYPE CODE 2  VEHICLE TYPE CODE 3 VEHICLE TYPE CODE 4  \
0                Sedan                  NaN                 NaN   
1                  NaN                  NaN                 NaN   
2        Pick-up Truck                  NaN                 NaN   
3                  NaN                  NaN                 NaN   
4                  NaN                  NaN                 NaN   

  VEHICLE TYPE CODE 5  
0                 NaN  
1                 NaN  
2                 NaN  
3                 NaN  
4                 NaN  

[5 rows x 29 columns]

In [20]:
df_crashes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1972121 entries, 0 to 1972120
Data columns (total 29 columns):
 #   Column                         Dtype  
---  ------                         -----  
 0   CRASH DATE                     object 
 1   CRASH TIME                     object 
 2   BOROUGH                        object 
 3   ZIP CODE                       object 
 4   LATITUDE                       float64
 5   LONGITUDE                      float64
 6   LOCATION                       object 
 7   ON STREET NAME                 object 
 8   CROSS STREET NAME              object 
 9   OFF STREET NAME                object 
 10  NUMBER OF PERSONS INJURED      float64
 11  NUMBER OF PERSONS KILLED       float64
 12  NUMBER OF PEDESTRIANS INJURED  int64  
 13  NUMBER OF PEDESTRIANS KILLED   int64  
 14  NUMBER OF CYCLIST INJURED      int64  
 15  NUMBER OF CYCLIST KILLED       int64  
 16  NUMBER OF MOTORIST INJURED     int64  
 17  NUMBER OF MOTORIST KILLED      int64  
 18  CO

In [26]:
# Estadistica descriptiva de las columas de personas fallecidas en accidentes
df_crashes[['NUMBER OF PERSONS KILLED','NUMBER OF PEDESTRIANS KILLED','NUMBER OF CYCLIST KILLED','NUMBER OF MOTORIST KILLED']].describe()

NUMBER OF PERSONS KILLED  NUMBER OF PEDESTRIANS KILLED  \
count              1.972090e+06                  1.972121e+06   
mean               1.412207e-03                  7.169945e-04   
std                3.959050e-02                  2.727384e-02   
min                0.000000e+00                  0.000000e+00   
25%                0.000000e+00                  0.000000e+00   
50%                0.000000e+00                  0.000000e+00   
75%                0.000000e+00                  0.000000e+00   
max                8.000000e+00                  6.000000e+00   

       NUMBER OF CYCLIST KILLED  NUMBER OF MOTORIST KILLED  
count              1.972121e+06               1.972121e+06  
mean               1.049631e-04               5.729872e-04  
std                1.029399e-02               2.607983e-02  
min                0.000000e+00               0.000000e+00  
25%                0.000000e+00               0.000000e+00  
50%                0.000000e+00               0.000000e+00  
75%                0.000000e+00               0.000000e+00  
max                2.000000e+00               5.000000e+00